workers/data_refinery_workers/processors/utils.py

```diff
@@ -668,3 +670,48 @@ def handle_processor_exception(job_context, processor_key, ex):
     job_context["job"].failure_reason = err_str
     job_context["success"] = False
     return job_context
+
+
+def cache_keys(*keys, work_dir_key='work_dir'):
+    """ Decorator to be applied to a pipeline function.
+    Returns a new function that calls the original one and caches the given
+    keys into the `work_dir`. On the next call it will load those keys (if they
+    exist) and add them to the job_context instead of executing the function. """
+    def inner(func):
+        key_names = ''.join(list(keys) + [func.__name__])
+        # generate a unique name for the cache based on the pipeline name
+        # and the cached keys
+        cache_name = hashlib.md5(key_names.encode()).hexdigest()
+
+        def pipeline(job_context):
+            cache_path = os.path.join(job_context[work_dir_key], cache_name)
+            if os.path.exists(cache_path):
+                # cached values exist, load keys from cacke
+                try:
+                    values = pickle.load(open(cache_path, "rb" ))
+                    return {**job_context, **values}
```

Comment:
Any reason to not just [update](https://docs.python.org/3/library/stdtypes.html#dict.update) `job_context` and return it?

Reply:
Not really, just an old habit of not modifying the parameters. Do you think this has any disadvantages?

Reply:
> just an old habit of not modifying the parameters

I do generally love this. `job_context` gets mutated directly all over the place though. We can leave it like this, works for me.